### PR TITLE
[Core] expose transpose matrix to python and add test

### PR DESCRIPTION
**Pull request by VeronikaSinger**
Hi,

this PR should expose the transpose function for a matrix to python.
A corresponding python test is added, too.

### CODE REVIEW

File: kratos/python/add_matrix_to_python.cpp [hunk 1]
```diff
@@ -97,6 +97,7 @@ namespace Python
         matrix_binder.def(py::init<const DenseMatrix<double>::size_type, const DenseMatrix<double>::size_type, const DenseMatrix<double>::value_type >());
         matrix_binder.def("fill", [](DenseMatrix<double>& self, const typename DenseMatrix<double>::value_type value) { noalias(self) = DenseMatrix<double>(self.size1(),self.size2(),value); });
         matrix_binder.def("fill_identity", [](DenseMatrix<double>& self) { noalias(self) = IdentityMatrix(self.size1()); });
+        matrix_binder.def("transpose", [](DenseMatrix<double>& self) { return Matrix(trans(self)); });
```
Comment (philbucher) [inline kratos/python/add_matrix_to_python.cpp:100]:
```suggestion
        matrix_binder.def("transpose", [](DenseMatrix<double>& self) { return trans(self); });
```

do you need the additional copy?

File: kratos/tests/test_matrix_interface.py [hunk 2]
```diff
@@ -101,5 +101,17 @@ def test_scalar_prod(self):
...
+        A = KM.Matrix(2,3)
+        B = KM.Matrix(3,2)
+        for i in range(A.Size1()):
+            for j in range(A.Size2()):
+                A[i,j] = i
+        B = A.transpose()
+
+        for i in range(A.Size1()):
+            for j in range(A.Size2()):
+                self.assertEqual(A[i,j], B[j,i])
```
Comment (philbucher) [inline kratos/tests/test_matrix_interface.py:114]:
```suggestion
        self.assertMatrixAlmostEqual(A, B)
```

**Review by philbucher — COMMENTED** [review]
thx!

File: kratos/python/add_matrix_to_python.cpp (hunk 1 above)
Comment (VeronikaSinger) [reply kratos/python/add_matrix_to_python.cpp:100]: otherwise i get an error

Comment (philbucher) [reply kratos/python/add_matrix_to_python.cpp:100]: can you post it?

File: kratos/tests/test_matrix_interface.py (hunk 2 above)
Comment (VeronikaSinger) [reply kratos/tests/test_matrix_interface.py:114]: the matrices are not equal as i and j are different

Comment (philbucher) [reply kratos/tests/test_matrix_interface.py:114]: sure
:man_facepalming:

File: kratos/python/add_matrix_to_python.cpp (hunk 1 above)
Comment (VeronikaSinger) [reply kratos/python/add_matrix_to_python.cpp:100]: TypeError: Unable to convert function return value to a Python type! The signature was
        (self: Kratos.Matrix) -> boost::numeric::ublas::matrix_unary2<boost::numeric::ublas::matrix<double,boost::numeric::ublas::basic_row_major<unsigned __int64,__int64>,boost::numeric::ublas::unbounded_array<double,std::allocator<double> > >,boost::numeric::ublas::scalar_identity<double> >

Did you forget to `#include <pybind11/stl.h>`? Or <pybind11/complex.h>,
<pybind11/functional.h>, <pybind11/chrono.h>, etc. Some automatic
conversions are optional and require extra headers to be included
when compiling your pybind11 module.

**Review by philbucher — APPROVED** [review]
ok thx

**Comment by VeronikaSinger** [comment]
thanks @philbucher 
Can I merge this or should I wait a few days?

**Comment by philbucher** [comment]
I think you can merge now :)

**Pull request merged**